lib/src/link_parser.dart

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'charcode.dart';
+import 'text_parser.dart';
+import 'util.dart';
+
+class LinkParser extends TextParser {
+  /// If there is a valid link formed.
+  bool valid = false;
```

Comment:
If these fields are not used by other types, please make them private!

Reply:
Seeing that they are used, then they should be getters. I know it's extra work, but we want to avoid exposing unwanted mutable state